sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -502,7 +502,8 @@ final class Decimal extends Ordered[Decimal] with Serializable {
     Decimal(toJavaBigDecimal.multiply(that.toJavaBigDecimal, MATH_CONTEXT))
 
   def / (that: Decimal): Decimal =
-    if (that.isZero) null else Decimal(toJavaBigDecimal.divide(that.toJavaBigDecimal, MATH_CONTEXT))
+    if (that.isZero) null else Decimal(toJavaBigDecimal.divide(that.toJavaBigDecimal,
+      DecimalType.MAX_SCALE, MATH_CONTEXT.getRoundingMode))
```

Comment:
Hive also use this API before 2.2:
https://github.com/apache/hive/blob/rel/release-2.1.1/storage-api/src/java/org/apache/hadoop/hive/common/type/HiveDecimal.java#L233-L235

Reply:
what does Hive use after 2.2?

Reply:
Implemented fast decimal since 2.2: https://issues.apache.org/jira/browse/HIVE-15335